documentcloud/entities/models.py

```diff
@@ -0,0 +1,75 @@
+from django.db import models
+from documentcloud.core.fields import AutoCreatedField, AutoLastModifiedField
+from wikidata.client import Client
+from django.utils.translation import gettext_lazy as _
+from .choices import EntityAccess
+
+
+class Entity(models.Model):
+    wd_entity = None
+
+    # A dictionary with language codes as keys.
+    name = models.CharField(max_length=500)
+    localized_names = models.JSONField()
+    wikidata_id = models.CharField(max_length=16)
+    # A dictionary with language codes as keys.
+    wikipedia_url = models.JSONField()
+    owner = models.ForeignKey(
+        "users.User", related_name="entities", on_delete=models.CASCADE
+    )
+    description = models.JSONField()
+    created_at = AutoCreatedField(
+        _("created at"),
+        db_index=True,
+        help_text=_("Timestamp of when the entity was created"),
+    )
+    updated_at = AutoLastModifiedField(
+        _("updated at"), help_text=_("Timestamp of when the entitywas last updated")
+    )
+    access = models.IntegerField(
+        _("access"),
+        choices=EntityAccess.choices,
+        help_text=_("Designates who may access this entity."),
+    )
+
+    def save(self, *args, **kwargs):
+        if not self.wikidata_id:
+            # Unnecessary?
+            raise "Missing wikidata_id in entity."
+
+        if not self.wikipedia_url:
+            self.wikipedia_url = self.get_url_for_wikidata_id()
+
+        if not self.name or not self.localized_names:
+            self.localized_names = self.get_names_for_wikidata_id()
+            # English bias here. TODO: How can this be addressed?
+            self.name = self.localized_names["en"]
+            if not self.name:
+                keys = self.localized_names.keys()
+                if len(keys) > 0:
+                    self.name = self.localized_names[keys[0]]
+                else:
+                    self.name = "Unknown"
+
+        if not self.description:
+            self.description = self.get_description_for_wikidata_id()
+
+        super().save(*args, **kwargs)
+
+    def get_wikidata_entity(self):
+        # TODO: Handle 404
```

Comment:
How is the performance of this in practice?  Should this be handled out of the request cycle in a celery task?

Pros of inline:
* can return an appropriate error if it doesn't exist
* can return the collected data in the response

Pros of celery:
* HTTP request can return faster (no external IO)
* Can retry on intermittent network issues

Reply:
It seems "good", as in I don't notice it taking any time for a single post, but it may not scale well. Maybe we can start with it doing the call inline, then when we do the entity upload project @morisy mentioned, we can go to celery if it's a problem?